cumulus/client/cirrus-executor/src/tests.rs

```diff
@@ -643,17 +643,15 @@ async fn pallet_executor_unsigned_extrinsics_should_work() {
 
 	// max drift is 4, hence the max allowed receipt number is 3 + 4, 8 will be rejected as being
 	// too far.
-	assert!(create_and_send_submit_execution_receipt(8).await.is_err());
-	// TODO: improve the pallet-executor error code to assert the specific error type.
-	// This also requires `sc_transaction_pool::error::Error` supports `==`. Should submit a PR to Substrate.
-	// assert_eq!(
-	// create_and_send_submit_execution_receipt(4).await.unwrap_err(),
-	// sc_transaction_pool::error::Error::Pool(
-	// sc_transaction_pool_api::error::Error::InvalidTransaction(
-	// pallet_executor::InvalidTransactionCode::ExecutionReceipt.into()
-	// )
-	// )
-	// );
+	let _execution_receipt_pool_error = sc_transaction_pool::error::Error::Pool(
```

Comment:
Tried to avoid it, but failed :)

```
   Compiling cirrus-client-executor v0.1.0 (/home/xlc/src/github.com/subspace/subspace/cumulus/client/cirrus-executor)
warning: unused variable: `execution_receipt_pool_error`
   --> cumulus/client/cirrus-executor/src/tests.rs:646:6
    |
646 |     let execution_receipt_pool_error = sc_transaction_pool::error::Error::Pool(
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_execution_receipt_pool_error`
    |
    = note: `#[warn(unused_variables)]` on by default

warning: unused variable: `execution_receipt_pool_error`
   --> cumulus/client/cirrus-executor/src/tests.rs:653:3
    |
653 |         execution_receipt_pool_error
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_execution_receipt_pool_error`
```

Reply:
Weird, what if you inline it in `assert_matches!()`? Looks like a compiler false-positive to me.

Reply:
doesn't compile if everything is inlined, still the unused warning if partially inlined. Probably hitting an edge case of this macro.

```diff
        assert_matches!(
                create_and_send_submit_execution_receipt(8).await.unwrap_err(),
-               _execution_receipt_pool_error
+               sc_transaction_pool::error::Error::Pool(
+                       sc_transaction_pool_api::error::Error::InvalidTransaction(
+                               pallet_executor::InvalidTransactionCode::ExecutionReceipt.into(),
+                       )
+               )
        );
```
```
   Compiling cirrus-client-executor v0.1.0 (/home/xlc/src/github.com/subspace/subspace/cumulus/client/cirrus-executor)
error: expected one of `!`, `(`, `)`, `,`, `...`, `..=`, `..`, `::`, `{`, or `|`, found `.`
   --> cumulus/client/cirrus-executor/src/tests.rs:655:62
    |
655 |                 pallet_executor::InvalidTransactionCode::ExecutionReceipt.into(),
    |                                                                          ^
    |                                                                          |
    |                                                                          expected one of 10 possible tokens
    |                                                                          help: missing `,`

error[E0531]: cannot find tuple struct or tuple variant `into` in this scope
   --> cumulus/client/cirrus-executor/src/tests.rs:655:63
    |
655 |                 pallet_executor::InvalidTransactionCode::ExecutionReceipt.into(),
    |                                                                           ^^^^ not found in this scope

error[E0023]: this pattern has 2 fields, but the corresponding tuple variant has 1 field
   --> cumulus/client/cirrus-executor/src/tests.rs:655:5
    |
654 |             sc_transaction_pool_api::error::Error::InvalidTransaction(
    |             ---------------------------------------------------------
655 |                 pallet_executor::InvalidTransactionCode::ExecutionReceipt.into(),
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ^^^^^^ expected 1 field, found 2
    |
   ::: /home/xlc/.cargo/git/checkouts/substrate-7e08433d4c370a21/c4f3d02/client/transaction-pool/api/src/error.rs:36:24
    |
36  |     InvalidTransaction(InvalidTransaction),
    |                        ------------------ tuple variant has 1 field

Some errors have detailed explanations: E0023, E0531.
For more information about an error, try `rustc --explain E0023`.
error: could not compile `cirrus-client-executor` due to 3 previous errors
```

Reply:
:confused:

Reply:
Regular match might have been better then :upside_down_face:

Reply:
Make sense, updated.